prediction_market_agent/agents/microchain_agent/nft_treasury_game/deploy_nft_treasury_game.py

```diff
@@ -84,28 +94,72 @@ def load(self) -> None:
 
         super().load()
 
+    @classmethod
+    def get_initial_system_prompt(cls) -> str:
+        system_prompt = nft_treasury_game_base_prompt(wallet_address=cls.wallet_address)
+        if cls.role == Role.buyer:
+            system_prompt += nft_treasury_game_buyer_prompt()
+        elif cls.role == Role.seller:
+            system_prompt += nft_treasury_game_seller_prompt()
+        return system_prompt
+
+    def get_holding_n_nft_keys(self) -> int:
+        contract = ContractOwnableERC721OnGnosisChain(address=NFT_TOKEN_FACTORY)
```

Comment:
```suggestion
        contract = ContractERC721BaseClass(address=NFT_TOKEN_FACTORY)
```

Reply:
Suggesting the above because no need for it to be `ownable` - just the ERC721 component has `balanceOf` functionality

Reply:
But it is ownable https://github.com/gnosis/labs-contracts/blob/dd748a506c9ee0cdc90778417b2743cc82e26aeb/src/NFT/AgentNFT.sol#L8. The idea for these classes is to be 1:1 with an underlying contract.